components/dashboard/public/complete-auth/index.html

```diff
@@ -6,21 +6,31 @@
 -->
 
 <html>
+
 <head>
     <meta charset='utf-8'>
     <title>Done</title>
     <script>
         if (window.opener) {
             const search = new URLSearchParams(window.location.search);
             const message = search.get("message");
-            window.opener.postMessage(message, `https://${window.location.hostname}`);
+            const newUser = search.get("newUser");
+            window.opener.postMessage(message + "&newUser=" + newUser, `https://${window.location.hostname}`);
+
+            if (newUser === "true") {
+                localStorage.setItem("newUserOnboardingPending", newUser);
```

Comment:
we use this localStorage item as a token, which the workspace list redeems in exchange for a modal with the organization join message.